### PR TITLE
Payouts: Add payout bank reference key to payout reports and CSV

### LINKED ISSUE #5316: Allow reconciliation of bank statements & daily deposits - e.g. show deposit `Payout trace ID` in Woo admin dashboard, or other unique ID to link
### Description

In 5789782-zen a new merchant with a fairly busy site has asked if there is any way to provide additional information to allow easy reconciliation of bank statements to deposits.

The bank statement contains the `Payout trace ID`.

The `Payout trace ID` is not available in reporting on the site, nor is it available via the `payout.paid` event. But would be quite helpful for merchants' accounts departments trying to match up daily deposits with their bank statements, especially considering that pending periods sometimes make matching by date alone tricky.

### Additional information ( added by @nagpai )

* Related changes needed on the server - https://github.com/Automattic/transact-platform-server/issues/6838 . This will need to be addressed to support changes for the client-side changes.

### PR DESCRIPTION
**Pull request by nagpai**
Fixes #5316 

>[!NOTE]
>  Note added after PR approval - for future reference - `Trace ID` was renamed to `Bank reference key`, based on [review feedback](https://github.com/Automattic/transact-platform-server/pull/6917#pullrequestreview-2468911647) and p1732861507747189/1732857021.096589-slack-C02BW3Z8SHK. If you spot trace ID anywhere in the discussion below ( other than Stripe Payout object ), please read it as `Bank reference key`

#### Changes proposed in this Pull Request

* Add `Trace ID` column in payouts list seen within `Payments > Payouts` , and within CSV exports. The `Trace ID` will help merchants match inbound payout entries in their bank statements with WooPayments Payouts. 
* This will be visible for all Payouts after November 21, 2024

#### Testing instructions

* Please check out your local server to the branch - `add/6838-payout-trace-id` , unless it is already merged. 
* Do a `npm run down && npm run up` on the server to restart docker and make sure the `trace_id` column is added to the payouts cache tables. 
* Check in to this branch on the client
* Make sure your linked account on the client has a few payouts after November 21, 2024.
* You may link to `acct_1OuYn1QpUShfetUA` if you like, since that account has a few sample payouts that can be helpful for testing. 
* Browse to `Payments > Deposits` 
* You should see the `Trace ID` column and the ID on the payouts after November 21, 2024.

 

##### Test browser export
* On the Payouts report page, use a filter to have payouts less than 25 and click on download
* Download the export file, and check the Trace ID's appearing in the report

##### Test endpoint export
* Make sure you have more than 25 payouts in your account
* On the Payouts report page, click on the download button.
* You should see a message that the report would be emailed to you
* Login to the wp-admin of the local server
* Check the email log and click the download link in the email you see there for the export
* You should see the Trace ID column with the data correctly in the CSV file

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9832` or branch name `add/5316-payout-trace-id` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/5316-payout-trace-id) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 1d12f10e247860779332ce8f03a7b1eff509206d
- Build time: 2024-11-29 09:27:40 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/deposits/list/index.tsx [hunk 1]
```diff
@@ -97,6 +97,11 @@ const getColumns = ( sortByDate?: boolean ): DepositsTableHeader[] => [
 		screenReaderLabel: __( 'Bank account', 'woocommerce-payments' ),
 		isLeftAligned: true,
 	},
+	{
+		key: 'traceId',
+		label: __( 'Trace ID', 'woocommerce-payments' ),
+		screenReaderLabel: __( 'Trace ID', 'woocommerce-payments' ),
```
Comment (brucealdridge) [inline client/deposits/list/index.tsx]: We should consider changing this to something that is clearer to Merchants such as `Reference`

**Review by brucealdridge — APPROVED** [review]
✅ Works as expected.

It would be good to add the `trace_id` to the payout details page but that would need design input and shouldn't hold up this PR.

**Comment by nagpai** [comment]
Marking as blocked until the [required server-side PR](https://github.com/Automattic/transact-platform-server/pull/6917) is merged.

**Comment by nagpai** [comment]
Merging this PR slightly before the server, after making sure it does not break anything - other than showing `N/A` in `Bank reference key` column - this gets fixed once the server PR is deployed on Monday. ( Today is Friday ) 

Merging it early so that it can catch the 8.6.0 release. 

Discussed here: p1732879431882379-slack-CGGCLBN58

**Pull request merged**